### PR TITLE
Configure CI for docker

### PR DESCRIPTION
**Pull request by FarzanRashid**
This PR updates github workflow to upload image on Dockerhub.

### CODE REVIEW

**Review by FarsanRashid — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,3 +53,25 @@ jobs:
 
     - name: Check code style with black
       run: tox -e black
+  build-and-push-docker-image-on-merge:
+    runs-on: ubuntu-latest
+    needs: [ test, static_analysis ]
+
+    if: github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged && github.event.pull_request.base.ref == 'main'
+
```
[the remark this answers is not in the document]
Comment (FarzanRashid) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml
[the remark this answers is not in the document]
Comment (FarzanRashid) [reply .github/workflows/ci.yml:1]: Fixed.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,3 +53,25 @@ jobs:
 
     - name: Check code style with black
       run: tox -e black
+  build-and-push-docker-image-on-merge:
+    runs-on: ubuntu-latest
+    needs: [ test, static_analysis ]
+
+    if: github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged && github.event.pull_request.base.ref == 'main'
```
[the remark this answers is not in the document]
Comment (FarzanRashid) [reply .github/workflows/ci.yml]: Removed.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -53,3 +53,25 @@ jobs:
 
     - name: Check code style with black
       run: tox -e black
+  build-and-push-docker-image-on-merge:
```
[the remark this answers is not in the document]
Comment (FarzanRashid) [reply .github/workflows/ci.yml:57]: Done.

**Pull request merged**